doc/long_tutorials/handle_drift/plot_handle_drift.py

```diff
@@ -0,0 +1,367 @@
+
+The :py:func:`~spikeinterface.preprocessing.correct_motion()`
+function implements motion correction algorithms in a modular way
+wrapping a number of subfunctions that together implement the
+full drift correction algorithm.
+
+These drift-correction modules are:
+
+| **1.** ``localize_peaks()`` (detect spikes and localize their position on the probe)
+| **2.** ``select_peaks()`` (optional, select a subset of peaks to use to estimate motion)
+| **3.** ``estimate_motion()`` (estimate motion using the detected spikes)
+| **4.** ``interpolate_motion()`` (perform interpolation on the raw data to account for the estimated drift).
+
+All these sub-steps have many parameters which dictate the
+speed and effectiveness of motion correction. As such, ``correct_motion``
+provides three setting 'presets' which configure the motion correct
+to proceed either as:
+
+* **rigid_fast** - a fast, not particularly accurate correction assuming rigid drift.
+* **kilosort-like** - Mimics what is done in Kilosort.
+* **nonrigid_accurate** - A decentralized drift correction (DREDGE), introduced by the Paninski group.
+
+When using motion correction in your analysis, please make sure to
+:ref:`cite the appropriate paper for your chosen method<cite-motion-correction>`.
+
+
+**Now, let's dive into running motion correction with these three
+methods on a simulated dataset.**
+
+"""
+
+# %%
+# -------------------------------------------
+# Setting up and preprocessing the recording
+# -------------------------------------------
+#
+# First, we will import the modules we will need for this tutorial:
+
+import matplotlib.pyplot as plt
+import spikeinterface.full as si
+from spikeinterface.generation.drifting_generator import generate_drifting_recording
+from spikeinterface.preprocessing.motion import motion_options_preset
+from spikeinterface.sortingcomponents.motion_interpolation import correct_motion_on_peaks
+from spikeinterface.widgets import plot_peaks_on_probe
+
+# %% read the file
+# Next, we will generate a synthetic, drifting recording. This recording will
+# have 100 separate units with firing rates randomly distributed between
+# 15 and 25 Hz.
+#
+# We will create a zigzag drift pattern on the recording, starting at
+# 100 seconds and with a peak-to-peak period of 100 seconds (so we will
+# have 9 zigzags through our recording). We also add some non-linearity
+# to the imposed motion.
+
+# %%
+#.. note::
+#    This tutorial can take a long time to run with the default arguments.
+#    If you would like to run this locally, you may want to edit ``num_units``
+#    and ``duration`` to smaller values (e.g. 25 and 100 respectively).
+#
+#    Also note, the below code uses multiprocessing. If you are on Windows, you may
+#    need to place the code within a  ``if __name__ == "__main__":`` block.
+
+
+num_units = 10# 200  # 250 still too many I think!
+duration = 50  # 1000
```

Comment:
Guessing this was for your local build - should uncomment 200 and 1000.

Reply:
Good spot! have updated